pkg/apis/garden/v1beta1/types.go

```diff
@@ -1250,6 +1250,22 @@ type CloudControllerManagerConfig struct {
 // KubeControllerManagerConfig contains configuration settings for the kube-controller-manager.
 type KubeControllerManagerConfig struct {
 	KubernetesConfig `json:",inline"`
+	// HorizontalPodAutoscalerConfig contains horizontal pod autoscaler configuration settings for the kube-controller-manager.
+	// +optional
+	HorizontalPodAutoscalerConfig *HorizontalPodAutoscalerConfig `json:"horizontalPodAutoscaler,omitempty"`
+}
+
+// HorizontalPodAutoscalerConfig contains horizontal pod autoscaler configuration settings for the kube-controller-manager.
+// Note: Descriptions were taken from the Kubernetes documentation.
+type HorizontalPodAutoscalerConfig struct {
```

Comment:
I did think about it and decided against it because there are multiple levels of conditional defaulting at play here.

1. Kube-controller-manager defaults.
2. Helm chart defaults. We override some defaults such as the `downscaleDelay` and `upscaleDelay`.
3. Defaults for the seed as overridden in the hyprid botanist. Admittedly, this is a problem only for Gardener and its operators.
4. Golang defaults. These would be not applicable if we use pointers as in the OIDC approach, of course.

I thought figuring out which ones are applicable in which situations might be too confusing while specifying it in the `Shoot` resource. Perhaps it is better that if someone wants to overriding the HPA configuration in the `Shoot` resource let them specify the whole thing and avoid any confusion.

WDYT?

Reply:
Hm, still, I would make every field configurable. It's our decision how we default them in the end (your 1. and 2. point), but we should not force the user to look this up.
The 3. point is, as you said, not of any concern. The 4. point: We would not default these values in the spec, we would just use the defaults we have configured in the Helm chart values.

Reply:
On the otherhand, I would argue it is all the more important to know what are the defaults if all the fields are optional because how can anyone be sure the values do not interact in problematic ways.

But anyway, I will make them configurable. Users can worry about it themselves ;-)

Reply:
Done

Reply:
> I would argue it is all the more important to know what are the defaults if all the fields are optional because how can anyone be sure the values do not interact in problematic ways.

That's true, maybe we should default them in the `pkg/apis/garden/v1beta1` to the actual default values we are using later (instead of putting them in the chart values). WDYT?
However, that's not related to what we had done here: We should not force the user to specify all values, hence, thank's for making them optional. Whether we default them ourselves is a different question.

Reply:
Is there a good way to default values in `pkg/apis/garden/v1beta1`? That might make it easy for users to find.

Reply:
Thanks! But what about the overriding of the downscaleDelay to 24h in the hybridbotanist? Shall we do it blindly for seeds regardless of whether the values came from actual `Shoot` resource or from the defaulting?

Reply:
Please remove [this](https://github.com/gardener/gardener/blob/master/pkg/operation/hybridbotanist/controlplane.go#L316-L319) code. We will configure the `Shoot` resources of the shooted Seeds to use these values. Now that we can configure it via the `Shoot` resource we do not longer need to "hard-code" it in our code. :-)

Reply:
Perfect! Will do.

Reply:
Done.

Where to change the Shoot specs for the seeds? If this is merged then seeds will go back to `downscaleDelay` of `15m`!

Reply:
This needs to be done once the new Gardener version gets deployed and can't be done in this repository. The change should be mentioned in the release note.

Thank you, I'll take another look to the PR asap.

Reply:
Ah, then coordination would be super important. Otherwise, after the new release gets deployed all seeds might go back to `15m` `downscaleDelay`!